drivers/wdm/audio/backpln/portcls/pin_wavert.cpp

```diff
@@ -305,35 +305,46 @@ CPortPinWaveRT::DeviceIoControl(
     IoStack = IoGetCurrentIrpStackLocation(Irp);
 
 
-    if (IoStack->Parameters.DeviceIoControl.IoControlCode == IOCTL_KS_PROPERTY)
+    switch (IoStack->Parameters.DeviceIoControl.IoControlCode)
     {
-       return HandleKsProperty(Irp);
+        case IOCTL_KS_PROPERTY:
+            return HandleKsProperty(Irp);
+            break;
+		
+        case IOCTL_KS_ENABLE_EVENT:
+            /* FIXME UNIMPLEMENTED */
+            UNIMPLEMENTED_ONCE;
+            break;
+
+        case IOCTL_KS_DISABLE_EVENT:
+            /* FIXME UNIMPLEMENTED */
+            UNIMPLEMENTED_ONCE;
+            break;
+
+        case IOCTL_KS_HANDSHAKE:
+            /* FIXME UNIMPLEMENTED */
+            UNIMPLEMENTED_ONCE;
+            break;
```

Comment:
In the limited scope of this PR, the 2 new cases should still use `return KsDefaultDeviceIoCompletion(DeviceObject, Irp);` instead.
Maybe with a `// TODO` to later double-check which code/status should be actually executed/returned.

Reply:
Oh, _that's_ where the difference is. Yes, at least for `IOCTL_KS_METHOD` it's important to call `KsDefaultDeviceIoCompletion`. For the other it shouldn't matter, but not changing behavior is good.